Jenkinsfile

```diff
@@ -0,0 +1,143 @@
+#!groovy
+/*
+ * This Jenkinsfile is intended to run on https://ci.jenkins-ci.org and may fail anywhere else.
+ * It makes assumptions about plugins being installed, labels mapping to nodes that can build what is needed, etc.
+ */
+
+// TEST FLAG - to make it easier to turn on/off unit tests for speeding up access to later stuff.
+def runTests = true
+
+// Only keep the 10 most recent builds.
+properties([[$class: 'BuildDiscarderProperty', strategy: [$class: 'LogRotator',
+                                                          numToKeepStr: '10']]])
+
+// TODO: Once https://github.com/jenkinsci/packaging/pull/34 is merged, switch to master branch.
+String packagingBranch = (binding.hasVariable('packagingBranch')) ? packagingBranch : 'oss-dockerized-tests'
+
+node('java') {
+
+    // Add timestamps to logging output.
+    wrap([$class: 'TimestamperBuildWrapper']) {
+
+        // First stage is actually checking out the source. Since we're using Multibranch
+        // currently, we can use "checkout scm".
+        stage "Checkout source"
+
+        // This will not work for standalone Workflow scripts. There, use:
```

Comment:
Per my discussion with @jglick this morning around [JENKINS-31386](https://issues.jenkins-ci.org/browse/JENKINS-31386), this comment is now incorrect (yey)